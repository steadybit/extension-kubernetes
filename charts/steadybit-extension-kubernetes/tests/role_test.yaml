templates:
  - role.yaml
tests:
  - it: manifest should match snapshot
    set:
      role:
        create: true
      roleBinding:
        create: true
      clusterRole:
        create: false
      clusterRoleBinding:
          create: false
    asserts:
      - matchSnapshot: { }
  - it: manifest should match snapshot with some disabled features
    set:
      role:
        create: true
      roleBinding:
        create: true
      clusterRole:
        create: false
      clusterRoleBinding:
        create: false
      discovery:
        disabled:
          cluster: true
          container: true
          daemonSet: true
          deployment: true
          ingress: true
          node: true
          pod: true
          replicaSet: false #default is true
          statefulSet: true
    asserts:
      - matchSnapshot: { }
