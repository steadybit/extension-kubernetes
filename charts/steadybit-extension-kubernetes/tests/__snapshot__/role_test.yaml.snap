manifest should match snapshot:
  1: |
    apiVersion: rbac.authorization.k8s.io/v1
    kind: Role
    metadata:
      labels: null
      name: steadybit-extension-kubernetes
      namespace: NAMESPACE
    rules:
      - apiGroups:
          - apps
        resources:
          - deployments
          - replicasets
          - daemonsets
          - statefulsets
        verbs:
          - get
          - list
          - watch
      - apiGroups:
          - ""
        resources:
          - services
          - pods
          - nodes
          - events
          - namespaces
        verbs:
          - get
          - list
          - watch
      - apiGroups:
          - autoscaling
        resources:
          - horizontalpodautoscalers
        verbs:
          - get
          - list
          - watch
      - apiGroups:
          - apps
        resources:
          - deployments
        verbs:
          - patch
      - apiGroups:
          - apps
        resources:
          - deployments/scale
        verbs:
          - get
          - update
          - patch
      - apiGroups:
          - apps
        resources:
          - statefulsets/scale
        verbs:
          - get
          - update
          - patch
      - apiGroups:
          - ""
        resources:
          - pods
        verbs:
          - delete
      - apiGroups:
          - ""
        resources:
          - pods/exec
        verbs:
          - create
manifest should match snapshot with some disabled features:
  1: |
    apiVersion: rbac.authorization.k8s.io/v1
    kind: Role
    metadata:
      labels: null
      name: steadybit-extension-kubernetes
      namespace: NAMESPACE
    rules:
      - apiGroups:
          - apps
        resources:
          - deployments
          - replicasets
          - daemonsets
          - statefulsets
        verbs:
          - get
          - list
          - watch
      - apiGroups:
          - ""
        resources:
          - services
          - pods
          - nodes
          - events
          - namespaces
        verbs:
          - get
          - list
          - watch
      - apiGroups:
          - autoscaling
        resources:
          - horizontalpodautoscalers
        verbs:
          - get
          - list
          - watch
      - apiGroups:
          - apps
        resources:
          - replicasets/scale
        verbs:
          - get
          - update
          - patch
